examples/hpc/dla-driver/src/lib.rs

```diff
@@ -18,291 +106,501 @@ macro_rules! set_bits {
     };
 }
 
-pub fn dla_write_str(s: &str) {
-    for b in s.as_bytes() {
-        unsafe { ptr::write_volatile(DLA0_ADDR as *mut u8, *b) };
-    }
-}
-pub fn dla_write(offset: usize, value: u8) {
-    unsafe { ptr::write_volatile((offset) as *mut u8, value) };
+macro_rules! get_bits {
+    ($reg:expr, $mask:expr) => {
+        ($reg & ($mask as u32)) as u32
+    };
 }
 
-pub fn dla_write_reg(offset: usize, value: u32) {
-    unsafe { ptr::write_volatile((DLA0_ADDR + offset) as *mut u32, value) }
-}
+impl Dla {
+    pub fn new() -> Self {
+        Dla {
+        }
+    }
+    pub fn write_str(&self, s: &str) {
+        for b in s.as_bytes() {
+            unsafe { ptr::write_volatile(DLA0_ADDR as *mut u8, *b) };
+        }
+    }
```

Comment:
Wait a second, what is this function achieving? It writes a string at the start of DLA0, is there a UART here or was this intended as a debug thing? Should this be removed?

Reply:
This was a leftover from the first version of dla driver, where it was used to write hello world to memory. Removed.